server/raft_transport.go

```diff
@@ -101,43 +115,101 @@ func (t *rpcTransport) Stop(id proto.RaftNodeID) {
 	delete(t.servers, id)
 }
 
-// Send a message to the recipient specified in the request.
-func (t *rpcTransport) Send(req *multiraft.RaftMessageRequest) error {
-	// Convert internal to proto formats.
-	protoReq := &proto.RaftMessageRequest{GroupID: req.GroupID}
-	var err error
-	if protoReq.Msg, err = req.Message.Marshal(); err != nil {
-		return err
+func (t *rpcTransport) processQueue(nodeID proto.RaftNodeID) {
+	t.mu.Lock()
+	ch, ok := t.queues[nodeID]
+	t.mu.Unlock()
+	if !ok {
+		return
 	}
-
-	nodeID, _ := proto.DecodeRaftNodeID(proto.RaftNodeID(req.Message.To))
-	addr, err := t.gossip.GetNodeIDAddress(nodeID)
-	if err != nil {
-		return err
+	// Clean-up when the loop below shuts down.
+	defer func() {
+		t.mu.Lock()
+		delete(t.queues, nodeID)
+		t.mu.Unlock()
+	}()
+	var req *multiraft.RaftMessageRequest
+	protoReq := &proto.RaftMessageRequest{}
+	for {
+		select {
+		case <-t.rpcContext.Stopper.ShouldStop():
+			return
+		case <-time.After(raftIdleTimeout):
+			return
+		case req = <-ch:
+		}
+		if req == nil {
+			continue
+		}
+
+		// Convert to proto format.
+		protoReq.Reset()
+		protoReq.GroupID = req.GroupID
+		var err error
+		if protoReq.Msg, err = req.Message.Marshal(); err != nil {
+			log.Errorf("could not marshal message: %s", err)
+		}
+
+		nodeID, _ := proto.DecodeRaftNodeID(proto.RaftNodeID(req.Message.To))
+		addr, err := t.gossip.GetNodeIDAddress(nodeID)
+		if err != nil {
+			log.Errorf("could not get address for node %d: %s", nodeID, err)
+		}
+
+		deadline := time.After(raftMessageTimeout)
+		client := rpc.NewClient(addr, nil, t.rpcContext)
```

Comment:
pulled it out as per our discussion. processQueue exits when the client fails anyways.